app/api/attendees.py

```diff
@@ -56,8 +60,13 @@ def before_post(self, args, kwargs, data):
                 "Ticket belongs to a different Event"
             )
         # Check if the ticket is already sold out or not.
-        if get_count(db.session.query(TicketHolder.id).
-                     filter_by(ticket_id=int(data['ticket']), deleted_at=None)) >= ticket.quantity:
+        order_expiry_time = get_settings()['order_expiry_time']
+        if get_count(db.session.query(TicketHolder.id)
+                    .filter(TicketHolder.ticket_id == int(data['ticket']),
+                            TicketHolder.state == 'placed', TicketHolder.deleted_at.is_(None),
+                            or_(TicketHolder.state ==' initializing', TicketHolder.created_at <
+                            datetime.datetime.utcnow()+datetime.timedelta(minutes=order_expiry_time))))\
```

Comment:
Why check the `order expiry time` condition? If the order is not created under it, Order is labelled expired. It is not required IMO.
Simply counting the `Placed & Completed` orders in the before the condition is enough.

Reply:
We need to count the currently reserved tickets. Or else more tickets than max tickets will be bought. See the discussion in the issue